dotCMS/src/main/java/com/dotcms/storage/DataBaseStoragePersistenceAPIImpl.java

```diff
@@ -0,0 +1,652 @@
+    public boolean existsObject(final String groupName, final String objectPath)
+            throws DotDataException {
+        final String groupNameLC = groupName.toLowerCase();
+        final String objectPathLC = objectPath.toLowerCase();
+        final MutableBoolean exists = new MutableBoolean(false);
+
+        try (Connection connection = getConnection()) {
+            final Number results = Try
+                    .of(() -> {
+                                final List<Map<String, Object>> result = new DotConnect()
+                                        .setSQL("SELECT count(*) as x FROM storage WHERE group_name = ? AND path = ?")
+                                        .addParam(groupNameLC).addParam(objectPathLC)
+                                        .loadObjectResults(connection);
+                                return (Number) result.get(0).get("x");
+                            }
+                    ).getOrElse(0);
+            exists.setValue(results.intValue() > 0);
+        } catch (SQLException e) {
+            throw new DotDataException(e.getMessage(), e);
+        }
+        ;
+
+        return exists.booleanValue();
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param groupName {@link String} group name
+     */
+    @Override
+    public boolean createGroup(final String groupName) throws DotDataException {
+        return createGroup(groupName, ImmutableMap.of());
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param groupName {@link String} group name
+     * @param extraOptions {@link Map} depending on the implementation it might need extra options
+     * or not.
+     */
+    @Override
+    public boolean createGroup(final String groupName, final Map<String, Object> extraOptions)
+            throws DotDataException {
+        final String groupNameLC = groupName.toLowerCase();
+        return wrapInTransaction(() ->
+                Try.of(() -> {
+                    new DotConnect().setSQL(" INSERT INTO storage_group (group_name) VALUES (?) ")
+                            .addParam(groupNameLC).loadResult(
+                            this.getConnection());
+                    return true;
+                }).getOrElseGet(throwable -> false)
+        );
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param groupName {@link String} group name
+     */
+    @Override
+    public int deleteGroup(final String groupName) throws DotDataException {
+        final String groupNameLC = groupName.toLowerCase();
+        return wrapInTransaction(() -> {
+            final Connection connection = this.getConnection();
+            final DotConnect dotConnect = new DotConnect();
+
+            final List<Map<String, Object>> maps = dotConnect
+                    .setSQL("SELECT hash FROM storage WHERE group_name = ?").addParam(groupNameLC)
+                    .loadObjectResults(connection);
+
+            int count = deleteObjects(maps.stream().map(map -> map.get("hash").toString())
```

Comment:
Codacy found an issue: [The String literal "hash" appears 4 times in this file; the first occurrence is on line 321](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)